src/fiber/network.rs

```diff
@@ -1162,6 +1162,33 @@ where
                                     )
                                     .await?;
                             }
+                            for tlc in actor_state.tlc_state.received_tlcs.get_committed_tlcs() {
+                                if let Some(payment_preimage) =
+                                    self.store.get_invoice_preimage(&tlc.payment_hash)
+                                {
+                                    debug!(
+                                        "Found payment preimage for channel {:?} tlc {:?}",
+                                        channel_id,
+                                        tlc.id()
+                                    );
+                                    let (send, _recv) = oneshot::channel();
+                                    let rpc_reply = RpcReplyPort::from(send);
+                                    state
+                                        .send_command_to_channel(
+                                            channel_id,
```

Comment:
is there a possibility that a channel both have a tlc with preimage and also expired tlcs?
maybe it is safer that we iterate channels twice, the first iteration is check tlc with preimage, the second iteration is check expired.